src/components/ReceiptEmptyState.tsx

```diff
@@ -18,13 +29,85 @@ type ReceiptEmptyStateProps = {
     disabled?: boolean;
 
     isThumbnail?: boolean;
+
+    shouldAllowReceiptDrop?: boolean;
+
+    transactionID: string | undefined;
 };
 
 // Returns an SVG icon indicating that the user should attach a receipt
-function ReceiptEmptyState({hasError = false, onPress, disabled = false, isThumbnail = false}: ReceiptEmptyStateProps) {
+function ReceiptEmptyState({hasError = false, onPress, disabled = false, isThumbnail = false, shouldAllowReceiptDrop = false, transactionID}: ReceiptEmptyStateProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const theme = useTheme();
+    const [isAttachmentInvalid, setIsAttachmentInvalid] = useState(false);
+    const [attachmentInvalidReasonTitle, setAttachmentInvalidReasonTitle] = useState<TranslationPaths>();
+    const [attachmentInvalidReason, setAttachmentValidReason] = useState<TranslationPaths>();
+    const [pdfFile, setPdfFile] = useState<null | FileObject>(null);
+    const [isLoadingReceipt, setIsLoadingReceipt] = useState(false);
+
+    /**
+     * Sets the upload receipt error modal content when an invalid receipt is uploaded
+     */
+    const setUploadReceiptError = (isInvalid: boolean, title: TranslationPaths, reason: TranslationPaths) => {
+        setIsAttachmentInvalid(isInvalid);
+        setAttachmentInvalidReasonTitle(title);
+        setAttachmentValidReason(reason);
+        setPdfFile(null);
+    };
+
+    const setReceipt = (originalFile: FileObject, isPdfValidated?: boolean) => {
+        validateReceipt(originalFile).then((result) => {
+            if (!result.isValid) {
+                if (result.title && result.reason) {
+                    setUploadReceiptError(true, result.title, result.reason);
+                }
+                return;
+            }
+
+            // If we have a pdf file and if it is not validated then set the pdf file for validation and return
+            if (Str.isPDF(originalFile.name ?? '') && !isPdfValidated) {
+                setPdfFile(originalFile);
+                return;
+            }
+
+            // With the image size > 24MB, we use manipulateAsync to resize the image.
```

Comment:
NAB but this comment can get outdated if we change the const value
```suggestion
            // With the image size > CONST.API_ATTACHMENT_VALIDATIONS.MAX_SIZE, we use manipulateAsync to resize the image.
```